infra/modules/common_environment/variables.tf

```diff
@@ -0,0 +1,58 @@
+#---------#
+# General #
+#---------#
+
+variable "tags" {
+  type        = map(any)
+  description = "Resources tags"
+}
+
+variable "environment" {
+  type = object({
+    prefix          = string
+    env_short       = string
+    location        = string
+    domain          = optional(string)
+    app_name        = string
+    instance_number = string
+  })
+
+  validation {
+    condition     = length("${var.environment.prefix}${var.environment.env_short}reg${var.environment.domain == null ? "" : replace(var.environment.domain, "-", "")}${var.environment.app_name}-apim-${var.environment.instance_number}") <= 50
+    error_message = "Azure API Management name must contain between 1 and 50 characters. Current value is \"${var.environment.prefix}${var.environment.env_short}reg${var.environment.domain == null ? "" : var.environment.domain}${var.environment.app_name}-apim-${var.environment.instance_number}\""
+  }
+
+  description = "Values which are used to generate resource names and location short names. They are all mandatory except for domain, which should not be used only in the case of a resource used by multiple domains."
+}
+
+variable "test_enable" {
```

Comment:
Is this intended to be turned on and off on daily basis?

Reply:
We discussed this previously. This variable is used to indicate whether the environment is a test environment or not. If it is a test environment, certain resources, such as the VPN and NAT, are not created and an extra resource group is provisioned to serve as a reference for creating resources specifically for testing purposes.

Reply:
Makes sense. But I recommend to rename it because it is grammatically incorrect